src/mlpack/methods/ann/ffn.hpp

```diff
@@ -142,6 +142,16 @@ class FFN
   void Predict(arma::mat& predictors, arma::mat& results);
 
   /**
+   * Predict the responses to a given set of predictors. The responses will
+   * reflect the output of the given output layer as returned by the
+   * output layer function.
+   *
+   * @param predictors Input predictors.
+   * @return Output predictions of responses into.
+   */
+  arma::mat Predict(arma::mat predictors);
+
```

Comment:
Hi @rcurtin , that's what exactly I mean. My point is we shouldn't allow `Predict(mat&)` and `Predict(const mat&)` exist together (So we need to delete current `Predict`). Instead we use `Predict(const mat&)` and `Predict(mat&&)` as you said.